marathon_lb.py

```diff
@@ -942,13 +1040,40 @@ def compareWriteAndReloadConfig(config, config_file):
     except IOError:
         logger.warning("couldn't open config file for reading")
 
-    if runningConfig != config:
-        logger.info(
-            "running config is different from generated config - reloading")
-        if writeConfigAndValidate(config, config_file):
-            reloadConfig()
-        else:
-            logger.warning("skipping reload: config not valid")
+    if haproxy_map:
+        if not os.path.isfile(map_file):
+            open(map_file, 'a').close()
+        runningmap = str()
+        try:
+            logger.debug("reading map config from %s", map_file)
+            with open(map_file, "r") as f:
+                runningmap = f.read()
+        except IOError:
```

Comment:
Hmm, so it continues if it fails to reload? What happens? Is there a possibility that HAProxy will be left in a weird state/